kani-driver/src/concrete_playback.rs

```diff
@@ -223,56 +155,152 @@ impl KaniSession {
     }
 }
 
-/// Generate a unit test from a list of concrete values.
+/// Writes the new source code to a temporary file. Returns whether the unit test was already in the old source code.
+fn write_new_src_to_tmp_file(
```

Comment:
I think this function signature is not very intuitive. I would probably create a struct TempFile that removes the file when dropping. You can create that in the beginning and return it to the caller to ensure it gets removed no matter what.